puppet/lib/puppet/indirector/catalog/puppetdb.rb

```diff
@@ -147,6 +148,17 @@ def add_code_id_if_missing(hash)
     hash
   end
 
+  # Include job_id in hash, returning the complete hash.
+  #
+  # @param hash [Hash] original data hash
+  # @return [Hash] returns original hash with a gaurunteed job_id key
+  # @api private
+  def add_job_id_if_missing(hash)
+    hash['job_id'] = hash['job_id']
```

Comment:
it sets the job id to nil if it doesn't exist already. it's silly code, but it's just copied from the function above for code_id.